cmd/api/v1/audit.go

```diff
@@ -0,0 +1,180 @@
+
+			cmd.Flags().String("from", "1h", "start of range of the audit traces. e.g. 1h, 10m, 2006-01-02 15:04:05")
+			cmd.Flags().String("to", "", "end of range of the audit traces. e.g. 1h, 10m, 2006-01-02 15:04:05")
+
+			cmd.Flags().String("user", "", "user of the audit trace.")
+			cmd.Flags().String("tenant", "", "tenant of the audit trace.")
+
+			cmd.Flags().String("project", "", "project id of the audit trace")
+
+			cmd.Flags().String("method", "", "api method of the audit trace.")
+			cmd.Flags().Int32("result-code", 0, "HTTP status code of the audit trace.")
+			cmd.Flags().String("source-ip", "", "source-ip of the audit trace.")
+
+			cmd.Flags().String("body", "", "filters audit trace body payloads for the giben text.")
+			cmd.Flags().String("error", "", "error of the audit trace.")
+
+			cmd.Flags().Int64("limit", 100, "limit the number of audit traces.")
+
+			genericcli.Must(cmd.RegisterFlagCompletionFunc("project", c.Completion.ProjectListCompletion))
+		},
+		DescribeCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.Flags().Bool("prettify-body", false, "attempts to interpret the body as json and prettifies it.")
+		},
+	}
+
+	return genericcli.NewCmds(cmdsConfig)
+}
+
+func (a *audit) Get(id string) (*apiv1.AuditTrace, error) {
+	ctx, cancel := a.c.NewRequestContext()
+	defer cancel()
+
+	project := a.c.GetProject()
+	fmt.Printf("project: %s", project)
+	tenant, err := a.c.GetTenant()
+	if err != nil {
+		return nil, fmt.Errorf("tenant is required")
+	}
+
+	req := &apiv1.AuditServiceGetRequest{
+		Login: tenant,
+		Uuid:  id,
+	}
+
+	resp, err := a.c.Client.Apiv1().Audit().Get(ctx, connect.NewRequest(req))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get audit trace: %w", err)
+	}
+
+	trace := resp.Msg.Audit
+
+	if viper.GetBool("prettify-body") {
+		if trace.Body != nil {
+			trimmed := strings.Trim(*trace.Body, `"`)
+			body := map[string]any{}
+			err = json.Unmarshal([]byte(trimmed), &body)
+			if err == nil {
+				if pretty, err := json.MarshalIndent(body, "", "    "); err == nil {
+					var prettifiedBody = string(pretty)
+					trace.Body = &prettifiedBody
+				}
+			}
+		}
+	}
+
+	return resp.Msg.Audit, nil
+}
+
+func (a *audit) List() ([]*apiv1.AuditTrace, error) {
+	ctx, cancel := a.c.NewRequestContext()
+	defer cancel()
+
+	fromDateTime, err := eventuallyRelativeDateTime(viper.GetString("from"))
+	if err != nil {
+		return nil, err
+	}
+	toDateTime, err := eventuallyRelativeDateTime(viper.GetString("to"))
+	if err != nil {
+		return nil, err
+	}
+
+	tenant, err := a.c.GetTenant()
+	if err != nil {
+		return nil, fmt.Errorf("tenant is required %w", err)
+	}
+
+	req := &apiv1.AuditServiceListRequest{
+		Login:      tenant,
+		Uuid:       pointer.PointerOrNil(viper.GetString("request-id")),
+		From:       fromDateTime,
+		To:         toDateTime,
+		User:       pointer.PointerOrNil(viper.GetString("user")),
+		Project:    pointer.PointerOrNil(viper.GetString("project")),
```

Comment:
Use project from config, if we want to allow requesting all projects we probably should add an explicit flag for this.